### DISCUSSION TITLE
Using a database with the fastify adapter

### DISCUSSION THREAD

**Discussion by vjau** [trpc/trpc#1998] [Q&A]
I'm trying trpc with the fastify adapter in a brownfield project.
My fastify server is using the mongo plugin which is creating a mongo connexion then is monkeypatching the Fastify server instance with a mongo field where you can find the db connexion.
How do you use this connexion in the trpc routers ?
The logical way would be to somehow get a grip on the connexion in the createContext function and add it to the context for the resolvers to consume, but i don't see a way to get access to the Fastify instance in createContext.
Is there a recommended way to do this ?

**Comment by KATT** [comment]
Hey! I don't use Fastify myself - @skarab42 could you help with this question?

**Comment by vjau** [comment]
Ok, it seems that the Fastify instance is on ```req.server```

**Reply by FabioJMV** [reply]
Thanks for that, I was wondering where the Fastify instance could be. However, I encountered a problem using [`@fastify/mysql`](https://github.com/fastify/fastify-mysql) plugin, I added the plugin to the Fastify context (_apparently_) succesfully, but then when I use the mysql object in the procedures it doesn't return anything but an empty object, so now I'm wondering if it could be a problem with the plugin or the trpc implementation, did you experience a similar problem using the mongo plugin?
